src/features/skills/skillsSlice.js

```diff
@@ -0,0 +1,111 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit"
+import axios from "axios"
+import instance from "../apiInstance"
+import { getErrorMessage } from "../../utils/getErrorMessage"
+import { ToastContainer } from "react-toastify"
+import { toast } from "react-toastify"
+
+const initialState = {
+  skills: null,
+  isError: false,
+  isSuccess: false,
+  isLoading: false,
+  message: "",
+}
+
+const CURRENT_SLICE_ROUTE = "/skills"
+// GET all skills
+export const getSkills = createAsyncThunk("getSkills", async (_, thunkAPI) => {
+  try {
+    return await instance.get(CURRENT_SLICE_ROUTE)
+  } catch (error) {
+    toast(error.message)
+    return thunkAPI.rejectWithValue(getErrorMessage(error))
+  }
+})
+
+// POST skill
+export const postSkill = createAsyncThunk(
+  "postSkill",
+  async (skill, thunkAPI) => {
+    try {
+      return await instance.post(CURRENT_SLICE_ROUTE, skill)
+    } catch (error) {
+      toast(error.message)
+      return thunkAPI.rejectWithValue(getErrorMessage(error))
+    }
+  }
+)
+
+// PUT update skill
+export const putSkill = createAsyncThunk(
+  "putSkill",
+  async ({ skillId, skillData }, thunkAPI) => {
+    try {
+      return await instance.put(`/skills/${skillId}`, skillData)
+    } catch (error) {
+      toast(error.message)
+      return thunkAPI.rejectWithValue(getErrorMessage(error))
+    }
+  }
+)
+
+// DELETE remove skill
+export const deleteSkill = createAsyncThunk(
+  "deleteSkill",
+  async (skillId, thunkAPI) => {
+    try {
+      return await instance.delete(`/skills/${skillId}`)
```

Comment:
Better concat the constant for current route ( `CURRENT_SLICE_ROUTE`) with `skillId`, same for `putSkill` request.